hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -433,23 +434,46 @@ private static Object convertValueForSpecificDataTypes(Schema fieldSchema, Objec
       return fieldValue;
     }
 
-    if (isLogicalTypeDate(fieldSchema)) {
-      return LocalDate.ofEpochDay(Long.parseLong(fieldValue.toString()));
+    if (fieldSchema.getType() == Schema.Type.UNION) {
+      for (Schema schema : fieldSchema.getTypes()) {
+        if (schema.getType() != Schema.Type.NULL) {
```

Comment:
Do we need to check this explicitly ? `convertValueForAvroLogicalTypes` will either ways just return the field value in this case.

The check is either ways not being performed in case Schema type is not `UNION`. It might be okay to remove this.

Reply:
For schema like:
```
{
"name":"event_cost3",
"type":[
"null",
{
"type":"fixed",
"name":"dc",
"size":5,
"logicalType":"decimal",
"precision":10,
"scale":6
}
]
```
or 
```
{
"name":"event_date3",
"type":[
"null",
{
"type":"int",
"logicalType":"date"
}
]
```

The first schema it gets is null, w/o this condition, it will directly return the fieldValue w/o any special conversion. BTW this comment remind me to add another test case for decimal type with union.